### DISCUSSION TITLE
Per layout keybinds

### DISCUSSION THREAD

**Discussion by tudorjnu** [qtile/qtile#4903] [Q&A] [answered]
I am trying to make some keybinds do different things in different layouts. One example that comes to mind is that in most layouts, mod+shift+<hjkl> result in moving the window. I want to be able to move a window similarly in the Tree layout. For example, mod+shift+j could move it down and mod+shift+l move it to the second section. Is there a way to implement this cleanly, or should I make a lazy function that follows:

get the layout
if layout is ... set keybind ...
else set keybind ...

**Comment by elParaguayo** [comment, marked as answer]
This is already possible. You can have multiple keybinds per key: `Key([mod], "h", lazy.layout..., lazy.layout..., lazy.layout...`.

To do layout specific actions, you do `Key([mod], "h", lazy.layout.swap_left().when(layout="monadtall")`

**Reply by tudorjnu** [reply]
I was absolutely not aware of this. Thank you!

**Reply by elParaguayo** [reply]
It's not well advertised in the docs but there are some examples here: https://docs.qtile.org/en/stable/manual/config/keys.html#defining-key-bindings

**Reply by tudorjnu** [reply]
No, it's my bad. It was there all along. Sorry for the unnecessary question, and thanks for this amazing project :pray:

**Reply by tudorjnu** [reply]
I am still having a problem with this setup, @elParaguayo. I am trying to set two keys to have different layout dependent functionalities. For example, I got:
```python
columns_keys = [
    Key([mod, "shift"], "Return", lazy.layout.toggle_split().when(layout="columns"), desc="Toggle between split and stack"),
    Key([mod, "control"], "h", lazy.layout.grow_left().when(layout="columns"), desc="Grow window to the left"),
    Key([mod, "control"], "l", lazy.layout.grow_right().when(layout="columns"), desc="Grow window to the right"),
    Key([mod, "control"], "j", lazy.layout.grow_down().when(layout="columns"), desc="Grow window down"),
    Key([mod, "control"], "k", lazy.layout.grow_up().when(layout="columns"), desc="Grow window up"),
]

monadtall_keys = [
    Key([mod, "control"], "l", lazy.layout.grow_main().when(layout="monadtall"), desc="Grow window"),
    Key([mod, "control"], "h", lazy.layout.shrink_main().when(layout="monadtall"), desc="Shrink window"),
]

```

Here, it seems that the keys are overridden by each other. Is there a way around this? Or am I making a mistake in the config?

**Reply by elParaguayo** [reply]
That's not going to work. You can only define a particular keybinding once. So, for example you'd need to do this for "l":
```python
Key(
    [mod, "control"], "l",
    lazy.layout.grow_right().when(layout="columns"),
    lazy.layout.grow_main().when(layout="monadtall")
)
```
When you press mod+ctrl+l, qtile will run every action that satisfies the conditions.

**Reply by tudorjnu** [reply]
Got it! Thank you!

**Comment by kaistian** [comment]
Since this question is very related, I hope it's OK to ask here to have it together.
Is there an easy way to say `not stack` in this `lazy.layout.left().when(layout=['stack'])`?
